cluster-commands.go

```diff
@@ -413,6 +462,14 @@ func (adm *AdminClient) SRPeerGetIDPSettings(ctx context.Context) (info IDPSetti
 	}
 
 	err = json.Unmarshal(b, &info)
+	if err != nil {
+		// If the server is older version, the IDPSettings was =
+		// LDAPSettings, so we try that.
+		err2 := json.Unmarshal(b, &info.LDAP)
+		if err2 == nil {
+			err = nil
+		}
+	}
```

Comment:
@donatello perhaps you can write a single UnmarshalJSON instead to support both cases instead of this kind of fallback. This way the struct has the right method for picking up the right struct.

thoughts?

Reply:
Not sure if it's a good idea - we only need the fallback unmarshal behavior in the specific case of supporting an older version server - there could be an issue in the future if `IDPSettings` is used in some other place.

Reply:
Okay @donatello